gcampus/auth/views/token.py

```diff
@@ -137,3 +125,30 @@ def logout(request: HttpRequest):
     django_logout(request)
     session.logout(request)
     return redirect("gcampuscore:mapview")
+
+
+class LoginView(TitleMixin, TemplateView):
+    template_name: str = "gcampusauth/login_form.html"
+    http_method_names = ["get"]
+    title = gettext_lazy("Login")
+
+    def get_context_data(self, **kwargs):
+        kwargs["access_key_form"] = AccessKeyForm()
+        kwargs["access_key_form_url"] = reverse("gcampusauth:login-access-key")
+        kwargs["course_token_form"] = CourseTokenForm()
+        kwargs["course_token_form_url"] = reverse("gcampusauth:login-course-token")
+        return super().get_context_data(**kwargs)
+
+
+class AccessKeyFormView(LoginFormView):
+    template_name: str = "gcampusauth/failed_login_form.html"
+    form_class = AccessKeyForm
+    success_url = reverse_lazy("gcampuscore:mapview")
+    token_type = TokenType.access_key
+
+
+class CourseTokenFormView(LoginFormView):
+    template_name: str = "gcampusauth/failed_login_form.html"
+    form_class = CourseTokenForm
+    success_url = reverse_lazy("gcampuscore:mapview")
```

Comment:
Maybe we could change the success_url to the course overview? I think that´s the page most people want to use after logging in with a course token?

Reply:
Thats a good idea